worker/caasoperator/caasoperator.go

```diff
@@ -210,6 +223,19 @@ func NewWorker(config Config) (worker.Worker, error) {
 		return nil, errors.Annotatef(err, "cannot create deployer")
 	}
 
+	podInit, err := newPodInitWorker(podInitConfig{
```

Comment:
I do think this needs to be a separate worker in the caasoperator manifold
And the NewWorker func is part of the manifold config so it can be tested etc

ie make the pod init worker and associated stuff look like the caasoperator worker
the can both run independently and as sibling workers of the top level runner
is there a core need to make this worker a child of the caas operator worker?

Reply:
@wallyworld can this be a part of a follow-up PR?

Reply:
never mind, I'm adding this